src/backend/base/langflow/components/tools/AstraCQLTool.py

```diff
@@ -0,0 +1,177 @@
+from langflow.io import DictInput, IntInput, SecretStrInput, StrInput
+from langflow.schema import Data
+
+
+class AstraCQLToolComponent(LCToolComponent):
+    display_name: str = "Astra DB CQL Tool"
+    description: str = "Create a tool to get data from DataStax Astra DB CQL Table"
+    documentation: str = "https://astra.datastax.com"
+    icon: str = "AstraDB"
+
+    inputs = [
+        StrInput(name="tool_name", display_name="Tool Name", info="The name of the tool.", required=True),
+        StrInput(
+            name="tool_description",
+            display_name="Tool Description",
+            info="The tool description to be passed to the model.",
+            required=True,
+        ),
+        StrInput(
+            name="keyspace",
+            display_name="Keyspace",
+            value="default_keyspace",
+            info="The keyspace name within Astra DB where the data is stored.",
+            required=True,
+        ),
+        StrInput(
+            name="table",
+            display_name="Table Name",
+            info="The name of the table within Astra DB where the data is stored.",
+            required=True,
+        ),
+        SecretStrInput(
+            name="token",
+            display_name="Astra DB Application Token",
+            info="Authentication token for accessing Astra DB.",
+            value="ASTRA_DB_APPLICATION_TOKEN",
+            required=True,
+        ),
+        StrInput(
+            name="api_endpoint",
+            display_name="API Endpoint",
+            info="API endpoint URL for the Astra DB service.",
+            value="ASTRA_DB_API_ENDPOINT",
+            required=True,
+        ),
+        StrInput(
+            name="projection",
+            display_name="Projection fields",
+            info="Attributes to return separated by comma.",
+            required=True,
+            value="*",
+        ),
+        DictInput(
+            name="partition_keys",
+            display_name="Partition Keys",
+            is_list=True,
+            info="Field name and description to the model",
+            required=True,
+        ),
+        DictInput(
+            name="clustering_keys",
+            display_name="Clustering Keys",
+            is_list=True,
+            info="Field name and description to the model",
+        ),
+        DictInput(
+            name="static_filters",
+            display_name="Static Filters",
+            is_list=True,
+            info="Field name and value. When filled, it will not be generated by the LLM.",
+        ),
+        IntInput(name="limit", display_name="Limit", value=5),
+    ]
+
+    def astra_rest(self, args):
+        headers = {"Accept": "application/json", "X-Cassandra-Token": f"{self.token}"}
+        ASTRA_URL = f"{self.api_endpoint}/api/rest/v2/keyspaces/{self.keyspace}/{self.table}/"
+        key = []
+        # Partition keys are mandatory
+        for k in self.partition_keys.keys():
```

Comment:
There is a make target, `make format` (and `make lint`) you can run to see if they catch these - we've made some ruff changes recently

Reply:
Ok. I ran format and lint locally.